squad/frontend/tests.py

```diff
@@ -85,15 +85,15 @@ def __get_page_filter__(self, page, per_page):
 
         stats = {}
         for test in self.all_tests:
-            if test.full_name not in stats:
-                stats[test.full_name] = {'pass': 0, 'fail': 0, 'xfail': 0, 'skip': 0, 'ids': []}
-            stats[test.full_name][test.status] += 1
-            stats[test.full_name]['ids'].append(test.id)
+            key = (test.suite_id, test.metadata_id)
+            if key not in stats:
+                stats[key] = {'pass': 0, 'fail': 0, 'xfail': 0, 'skip': 0, 'ids': []}
+            stats[key][test.status] += 1
+            stats[key]['ids'].append(test.id)
 
         def keyfunc(item):
-            name = item[0]
             statuses = item[1]
-            return tuple((-statuses[k] for k in ['fail', 'xfail', 'skip', 'pass'])) + (name,)
+            return tuple((-statuses[k] for k in ['fail', 'xfail', 'skip', 'pass']))
```

Comment:
same comment as above, with this we lose the ordering by name.

Reply:
ack